File: 559-maximum-depth-of-n-ary-tree/559-maximum-depth-of-n-ary-tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children
"""

class Solution:
    def maxDepth(self, root: 'Node') -> int:
        max_depth = 0
        if not root:
            return 0
        if not root.children:
            return 1
        for child in root.children:
            dep = 1 + self.maxDepth(child)
            max_depth = max(max_depth, dep)
        return max_depth

File: 559-maximum-depth-of-n-ary-tree/559-maximum-depth-of-n-ary-tree.cpp
/*
// Definition for a Node.
class Node {
public:
    int val;
    vector<Node*> children;

    Node() {}

    Node(int _val) {
        val = _val;
    }

    Node(int _val, vector<Node*> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
public:
    int maxDepth(Node* root) {
        int depth = 0;
        if (!root)
            return 0;
        if (root->children.empty())
            return 1;
        for (auto u: root->children) {
            depth = max(depth, maxDepth(u) + 1);
        }
        return depth;
    }
};